src/CodeGen/ODataT4CodeGenerator.ttinclude

```diff
@@ -3420,6 +3446,18 @@ namespace <#= fullNamespace #>
 
     internal override void WriteGeneratedEdmModel(string escapedEdmxString)
     {
+		string path = this.context.TempFilePath; 
+        
+		if(String.IsNullOrEmpty(path))
+		{
+			using (StreamWriter writer =
+            new StreamWriter(path, true))
+			{
+				writer.WriteLine(escapedEdmxString);
+			}
+		}
+
+		bool useTempFile = !String.IsNullOrEmpty(path) && System.IO.File.Exists(path);

```

Comment:
Please see my comment regarding the difference between TempFilePath and MetadataDocumentUri. Would it make sense have a useTempFile boolean that the user can set?

Reply:
Replied to the comment above and for the same reason, this should not be a boolean as we need a location where writes can be made.